test/state/mpt.cpp

```diff
@@ -219,46 +219,46 @@ void MPTNode::insert(const Path& path, bytes&& value)  // NOLINT(misc-no-recursi
     }
 }
 
-hash256 MPTNode::hash() const  // NOLINT(misc-no-recursion)
+bytes MPTNode::encode() const  // NOLINT(misc-no-recursion)
 {
+    static constexpr auto encode_child = [](const MPTNode& child) {  // NOLINT(misc-no-recursion)
+        if (auto e = child.encode(); e.size() < 32)
```

Comment:
```suggestion
        if (const auto e = child.encode(); e.size() < 32)
```

Reply:
This prevents move at `return e;`

Reply:
That's pretty weird, I think in a normal function copy/move would be elided even if returned variable is const. What if you define lamda's return type?

Reply:
Seems being under the `if` statement make a difference. https://godbolt.org/z/rq7W7fsxM